controller/controller.go

```diff
@@ -409,8 +418,12 @@ func (c *Impl) MaybeEnqueueBucketKey(bkt reconciler.Bucket, key types.Namespaced
 // the work queue after given delay.
 func (c *Impl) EnqueueKeyAfter(key types.NamespacedName, delay time.Duration) {
 	c.workQueue.AddAfter(key, delay)
-	c.logger.With(zap.String(logkey.Key, key.String())).
-		Debugf("Adding to queue %s (delay: %v, depth: %d)", safeKey(key), delay, c.workQueue.Len())
+
+	desugared := c.logger.Desugar()
+	if desugared.Core().Enabled(zapcore.DebugLevel) {
+		desugared.Debug(fmt.Sprintf("Adding to queue %s (delay: %v, depth: %d)", safeKey(key), delay, c.workQueue.Len()),
+			zap.String(logkey.Key, key.String()))
+	}
```

Comment:
Do we need to assign this to a new variable when we could just use `c.logger` (sugared) inside the block?

Reply:
Makes sense. You might want to assign the var in the `if` context if you don't use it anywhere else.
```suggestion
	if desugared := c.logger.Desugar(); desugared.Core().Enabled(zapcore.DebugLevel) {
		desugared.Debug(fmt.Sprintf("Adding to queue %s (delay: %v, depth: %d)", safeKey(key), delay, c.workQueue.Len()),
			zap.String(logkey.Key, key.String()))
	}
```